### PR TITLE
[#165410714] Fix wallet loading - credit card optional

### PR DESCRIPTION
**Pull request by Undermaken**
This PR aims to fix the wallets decoding when a wallet has no creditcard.
Before the creditcard field was mandatory, now it is optional.

### before
Simulator Screen Shot - iPhone X - 2019-06-13 at 20 06 05

### after
Simulator Screen Shot - iPhone X - 2019-06-13 at 20 01 10

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🐞 [#165410714](https://www.pivotaltracker.com/story/show/165410714): Impossibile caricare i metodi di pagamento se l'utente aveva già salvato dei metodi diversi dalle carte di credito.

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/components/wallet/card/CardComponent.tsx [hunk 1]
```diff
@@ -271,7 +271,7 @@ export default class CardComponent extends React.Component<Props> {
     const hasFlatBottom =
       this.props.type === "Preview" || this.props.type === "Header";
 
-    return (
+    return wallet.creditCard === undefined ? null : (
```
Comment (fimbles) [inline ts/components/wallet/card/CardComponent.tsx:274]: 1000

File: ts/components/wallet/card/CardComponent.tsx [hunk 2]
```diff
@@ -271,7 +271,7 @@ export default class CardComponent extends React.Component<Props> {
...
       <View
         style={[styles.card, hasFlatBottom ? styles.flatBottom : undefined]}
```
Comment (fimbles) [inline ts/components/wallet/card/CardComponent.tsx:276]: Fiat

File: ts/components/wallet/card/CardComponent.tsx [hunk 3]
```diff
@@ -295,7 +295,7 @@ export default class CardComponent extends React.Component<Props> {
             </View>
           </View>
 
-          {this.renderBody()}
+          {this.renderBody(wallet.creditCard)}
```
Comment (fimbles) [inline ts/components/wallet/card/CardComponent.tsx:298]: 1000

File: ts/components/wallet/card/Logo.tsx [hunk 4]
```diff
@@ -60,17 +60,13 @@ const styles = StyleSheet.create({
 });
 
 type Props = Readonly<{
-  item?: Wallet;
+  item?: CreditCard;
```
Comment (fimbles) [inline ts/components/wallet/card/Logo.tsx:63]: 1000

File: ts/types/__tests__/pagopa.test.ts [hunk 5]
```diff
@@ -122,6 +122,16 @@ const validWallet: { [key: string]: any } = {
   lastUsage: "2018-08-07T15:50:08Z"
 };
 
+const validWalletNoCreditCard: { [key: string]: any } = {
```
Comment (fimbles) [inline ts/types/__tests__/pagopa.test.ts:125]: 25/08/27

File: ts/types/__tests__/pagopa.test.ts [hunk 6]
```diff
@@ -122,6 +122,16 @@ const validWallet: { [key: string]: any } = {
...
+  idWallet: 2345,
+  type: "EXTERNAL_PS",
+  favourite: false,
```
Comment (fimbles) [inline ts/types/__tests__/pagopa.test.ts:128]: No

File: ts/types/__tests__/pagopa.test.ts [hunk 7]
```diff
@@ -231,6 +241,11 @@ describe("Wallet", () => {
   it("should accept a valid Wallet", () => {
     expect(Wallet.decode(validWallet).isRight()).toBeTruthy();
   });
+
+  it("should accept a valid Wallet that contains no credit card", () => {
```
Comment (fimbles) [inline ts/types/__tests__/pagopa.test.ts:245]: Si credit

**Pull request merged**